### PR TITLE
[Merged by Bors] - Generalize target type in fluivo-package-index

### PR DESCRIPTION
**Pull request by nicholastmosher**
This, followed by changes to `fluvio-packages`, will unblock us from publishing binaries from more targets than just `x86_64-unknown-linux-musl` and `x86_64-apple-darwin`.

### CODE REVIEW

File: src/package-index/src/target.rs [hunk 1]
```diff
@@ -14,36 +15,57 @@ pub fn package_target() -> Result<Target, Error> {
...
+///
+/// # Example
+///
+/// ```
+/// # use fluvio_index::Target;
+/// let target: Target = "x86_64-unknown-linux-musl".parse().unwrap();
+/// ```
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize)]
+#[serde(transparent)]
+pub struct Target(Cow<'static, str>);
```
Comment (simlay) [inline src/package-index/src/target.rs:32]: Hmm, I kinda like the enum of this better. I understand why you've done it this way though. Can one use the `transparent` and `rename` with serde at the sametime?

**Review by simlay — COMMENTED** [review]
This seems okay. I like enums more than a transparent string target. It might be worth fighting serde a little bit if you can use `rename` with a can work with a `transparent` container for say `Other(Cow<'static, str>)` as the unknown target type.

There's a few ownership changes that are unclear why they're needed but I'm okay with giving clippy and rustc the wheel.

File: src/package-index/src/target.rs (hunk 1 above)
Comment (nicholastmosher) [reply src/package-index/src/target.rs:32]: What is it about the enum that you like better? With the enum representation we would need to add variants for each new target we want to support. In either case, the "serialized" representation is the same, it is transparently represented as a string that is just the name of the target.

What do you mean about `rename`? I'm no longer using `rename` in this new version.

Comment (sehz) [reply src/package-index/src/target.rs:32]: @simlay ?

Comment (sehz) [reply src/package-index/src/target.rs:32]: Given that we are going to add more variants, I don't think we can encode as Enum

Comment (simlay) [reply src/package-index/src/target.rs:32]: Sounds good. Let's keep as is.

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**